src/components/IOUConfirmationList.js

```diff
@@ -244,34 +249,36 @@ class IOUConfirmationList extends Component {
         return (
             <>
                 <ScrollView style={[styles.flex1, styles.w100]}>
-                    <OptionsList
-                        listContainerStyles={[{
-                            // Give max height to the list container so that it does not extend
-                            // beyond the comment view as well as button
-                            maxHeight: this.props.windowHeight - MINIMUM_BOTTOM_OFFSET
-                                - this.props.insets.top - this.props.insets.bottom,
-                        }]}
-                        sections={this.getSections()}
-                        disableArrowKeysActions
-                        disableRowInteractivity
-                        hideAdditionalOptionStates
-                        forceTextUnreadStyle
-                        canSelectMultipleOptions={this.props.hasMultipleParticipants}
-                        disableFocusOptions
-                        selectedOptions={this.getAllOptionsAsSelected()}
-                    />
-                    <Text style={[styles.p5, styles.textMicroBold, styles.colorHeading]}>
-                        {this.props.translate('iOUConfirmationList.whatsItFor')}
-                    </Text>
-                    <View style={[styles.ph5, styles.pb5]}>
-                        <TextInput
-                            style={[styles.textInput]}
-                            value={this.props.comment}
-                            onChangeText={this.props.onUpdateComment}
-                            placeholder={this.props.translate('common.optional')}
-                            placeholderTextColor={themeColors.placeholderText}
+                    <Animatable.View animation={this.props.animation} duration={300}>
+                        <OptionsList
+                            listContainerStyles={[{
+                                // Give max height to the list container so that it does not extend
+                                // beyond the comment view as well as button
+                                maxHeight: this.props.windowHeight - MINIMUM_BOTTOM_OFFSET
+                                    - this.props.insets.top - this.props.insets.bottom,
+                            }]}
```

Comment:
We've manually implemented animations to these 4 files, but this is not easy to maintain or extend. Can you try to simplify this with the use of a new Component/HOC which can be reused by any page without needing to repeat the animation logic?

Reply:
I'm having a little trouble imagining this, can you help me here?

We're only wrapping the component to be animated in a `Animatable.View` and passing the animation and duration props, if we were to create a new Component/HOC, we would wrap the component in it and, if we wanted the animation and duration to be customizable, pass the same props as we did before.

I think I might be missing something here

Reply:
>  if we wanted the animation and duration to be customizable, pass the same props as we did before.

In this case, the props would not be passed. The animation duration and name would be defined once, within this component. It would mean that any other Modal can add the exact same page animations without needing to manually define props each time. But please let me know if I'm missing something that makes this tricky.

Something like this: 

```
<IOUModal
  <AnimatedStep animation='in|out'>
    <IOUAmountPage>
```

The benefits being **A)** We don't repeat ourselves by declaring the animation duration and type in multiple places **B)** that we have no dependency on animation logic in either the IOU pages or IOUModal. IOUModal simply declares: I want you to animate in -- keeping components nicely separated and reusable

Reply:
Ahh, I got it now. Thanks for the explanation!

I'll apply the changes and re-request the review once it's ready.